File: implementations/fastapi/examples/src/config/tracing_config.py
from pydantic_settings import BaseSettings, SettingsConfigDict


class TracingConfig(BaseSettings):
    model_config = SettingsConfigDict(env_prefix="OTEL_")

    # No default, and intentionally optional — unlike DATABASE_URL/JWT_SECRET, tracing must
    # not fail validate_env() when unset. Local dev/CI/tests run with no collector at all
    # (observability.md, "Metrics/tracing"): spans are still created (trace_id/traceparent
    # keep working end-to-end) but exported nowhere. Set OTEL_EXPORTER_OTLP_ENDPOINT to a
    # real collector's OTLP/HTTP endpoint (e.g. http://otel-collector:4318) to actually ship
    # spans in staging/production — the same sane-dev-default / real-value-required-in-prod
    # split used by JWT_SECRET/AwsConfig (config.md).
    exporter_otlp_endpoint: str | None = None
    service_name: str = "account-service"

File: implementations/nestjs/examples/src/config/tracing.config.ts
// Same "sane dev default, real value required in prod" shape as jwt.config.ts's JWT_SECRET /
// aws.config.ts's getAwsEndpoint — with no OTEL_EXPORTER_OTLP_ENDPOINT set (local dev/test),
// @/tracing exports spans to the console instead, so npm test/npm run start never need a real
// OpenTelemetry collector running. Set it to a real OTLP/HTTP collector endpoint (Jaeger,
// Tempo, a Datadog agent, etc.) in staging/production.
export function getOtlpEndpoint(): string | undefined {
  return process.env.OTEL_EXPORTER_OTLP_ENDPOINT
}

export function getServiceName(): string {
  return process.env.OTEL_SERVICE_NAME ?? 'account-service'
}
